src/js/angular/ttyg/controllers/agent-settings-modal.controller.js

```diff
@@ -224,6 +234,76 @@ function AgentSettingsModalController(
     // Private functions
     // =========================
 
+    /**
+     * Mapping of agent operations to their respective handlers.
+     * @type {{[AGENT_OPERATION.EDIT]: (function(*): Promise<void>), [AGENT_OPERATION.CLONE]: (function(*): Promise<void>), [AGENT_OPERATION.CREATE]: (function(*): Promise<void>)}}
+     */
+    const agentOperationHandlers = {
+        [AGENT_OPERATION.CREATE]: (payload) => createAgent(payload),
+        [AGENT_OPERATION.EDIT]: (payload) => editAgent(payload),
+        [AGENT_OPERATION.CLONE]: (payload) => cloneAgent(payload)
+    };
+
+    /**
+     * Creates a new agent with the given payload and closes the modal if no errors occur.
+     * @param {*} newAgentPayload - the payload for the new agent.
+     * @return {Promise<void>}
+     */
+    const createAgent = (newAgentPayload) => {
+        $scope.savingAgent = true;
+        return TTYGService.createAgent(newAgentPayload)
+            .then((agentModel) => {
+                $uibModalInstance.close(agentModel);
+                toastr.success($translate.instant('ttyg.agent.messages.agent_save_successfully', {agentName: agentModel.name}));
+            })
+            .catch((error) => {
+                toastr.error(getError(error, 0, 100));
```

Comment:
Maybe we should add an error message like in editAgent, instead of displaying what the backend returned.

Reply:
I will add one in my next MR